### PR TITLE
Make energy bola catchable again

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Title.

## Why It's Good For The Game

It being uncatchable sure as hell isn't.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

https://user-images.githubusercontent.com/22431091/159734917-5b612695-93eb-4038-a007-7d4b835f14c4.mp4

</details>

## Changelog
:cl:
fix: energy bola can be caught again
/:cl:

### CODE REVIEW

**Comment by Mat05usz** [comment]
Very nice change, but maybe make the ebola not disappear after being caught? I understand it's energy, but it should 'activate' only once it's fully around someone's legs, otherwise it should disappear after the throw, no matter if caught or not.

**Comment by ivanmixo** [comment]
It doesn't disappear when caught.

**Comment by Crossedfall** [comment]
Checks are stuck

**Review by PowerfulBacon — APPROVED** [review]
I thought this was already fixed, must have missed the energy ones

**Comment by ivanmixo** [comment]
It was fixed, but got broken by the crossed/uncrossed PR I think?

File: code/game/objects/items/handcuffs.dm [hunk 1]
```diff
@@ -377,12 +377,10 @@
...
-/obj/item/restraints/legcuffs/bola/energy/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
-	if(iscarbon(hit_atom))
-		var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(hit_atom))
-		B.spring_trap(null, hit_atom)
-		qdel(src)
-	. = ..()
+/obj/item/restraints/legcuffs/bola/energy/ensnare(mob/living/carbon/C)
+	var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(C))
+	B.spring_trap(null, C)
+	qdel(src)
```
Comment (Archanial) [inline code/game/objects/items/handcuffs.dm:383]: you are passing atom here but expect carbon

**Review by Archanial — DISMISSED** [review]

File: code/game/objects/items/handcuffs.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/handcuffs.dm:383]: This code seems fine to me

Comment (ivanmixo) [reply code/game/objects/items/handcuffs.dm:383]: Ehh yeah, I think this is fine. The passed atom will never not be a carbon and it's logged alright.

Comment (PowerfulBacon) [reply code/game/objects/items/handcuffs.dm:383]: spring_trap takes in an atom anyway

**Pull request merged**